cdap-api/src/main/java/co/cask/cdap/api/flow/FlowletConnection.java

```diff
@@ -34,9 +34,15 @@
   private final Type sourceType;
   private final String sourceName;
   private final String targetName;
+  private final String sourceNamespace;
 
   public FlowletConnection(Type sourceType, String sourceName, String targetName) {
+    this(sourceType, null, sourceName, targetName);
+  }
+
+  public FlowletConnection(Type sourceType, String sourceNamespace, String sourceName, String targetName) {
```

Comment:
Have discussed this in person. It is hard to make the subclass as FloweletConnection are wrapped in a List and encoded by gson to json string. The namespace field can either be null or String , which will get lost during the serialization/deserialization. Choose to keep it as now.